plugins/bookmarks/__init__.py

```diff
@@ -244,84 +230,73 @@ def delete_bookmark(self, widget, targets):
 
         self.save_db()
 
+    @common.idle_add()
```

Comment:
> Why is this idle_add?

Because it does I/O operations which might block, e.g. during swapping/paging or because of disk suspend.